### ISSUE TITLE
nx release publish launch 'npm dist-tag add' command on a package version which does not exists on the npm registry

### ISSUE THREAD

**Issue by Bare1337** [nrwl/nx#26240]
### Current Behavior

 During nx release publish execution and when there is only one version of the specified package on the remote registry, `npm dist-tag add ${packageName}@${packageJson.version} ${tag} --"${registryConfigKey}=${registry}"` is executed against a package version which does not exist in the registry.

In our case, it leads to an "unhandled" 400 error when it's executed with dist-tag "latest" on Sonatype Nexus Repository OSS 3.55.0-01.

### Expected Behavior

When the current package version to publish is not found on the remote registry, nx release publish should not try to execute the command `npm dist-tag add ${packageName}@${packageJson.version} ${tag} --"${registryConfigKey}=${registry}"` and error out.

### Steps to Reproduce

1. Create a new package in an nx workspace and update package.json version of the package to "0.1.0-beta.0"
2. Execute nx release publish --tag=latest
3. Update the created package package.json version to "0.1.0"
4. Execute nx release publish --tag=latest

### Nx Report

```shell
Node   : 20.9.0
OS     : win32-x64
pnpm   : 8.15.1

nx                 : 18.3.5
@nx/js             : 18.3.5
@nx/jest           : 18.3.5
@nx/linter         : 18.3.5
@nx/eslint         : 18.3.5
@nx/workspace      : 18.3.5
@nx/angular        : 18.3.5
@nx/cypress        : 18.3.5
@nx/devkit         : 18.3.5
@nx/eslint-plugin  : 18.3.5
@nrwl/tao          : 18.3.5
@nx/web            : 18.3.5
@nx/webpack        : 18.3.5
typescript         : 5.4.5
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

### Additional Information

The issue comes from the return type of the npm view command which is not consistent for the parsed "versions" key.

"versions" key return an array of string when there is more than one version published to the remote registry and a simple string while there is still only one version.

Hence if the versions key contains for example "0.1.0-beta.0" and i'am trying to publish "0.1.0", this part of the code:
https://github.com/nrwl/nx/blob/7088b09f472bb767d386df73ca0534e09386d7db/packages/js/src/executors/release-publish/release-publish.impl.ts#L113
 return true as it's a substring match.

We fixed it by preemptively adding a type check and reassigning the string value into an array before the check is executed.

I'am willing to submit a PR if necessary.

**Comment by JamesHenry** [comment]
Thank you for the write up @Bare1337, SMH at the different data type depending on the length... but I can indeed reproduce using a local verdaccio registry:

**After one publish of a package**

```
{ resultJson: { versions: '1.0.0', 'dist-tags': { latest: '1.0.0' } } }
```

**After two publishes of a package**

```
{
  resultJson: { versions: [ '1.0.0', '1.0.1' ], 'dist-tags': { latest: '1.0.1' } }
}
```

I will put up a quick PR now to normalize it on our side

**Comment by JamesHenry** [comment]
This will be resolved by #26496
